### PR TITLE
359 12 quick capture referral

### PR DESCRIPTION
**Pull request by rik86future**
**1. Save a Triage for Later**
Go to a triage
image

Click the 'Add Enquiry'
image

If the user click 'Save for later' this draft will be highlighted on the left nav bar
image

Also the user is redirected to the enquiry page
image

**2. Return to a triage**

It can be done, by clicking on the left nav bar or manually navigating to the correct triage.

The triage can now be completed or 'Discarded'

### CODE REVIEW

File: app/controllers/triage_controller.rb [hunk 1]
```diff
@@ -7,34 +7,60 @@ class TriageController < ApplicationController
...
+    end
   end
 
   def update
+    if params.require(:discard_draft) == 'true'
+      session[:triage] = nil
+      redirect_to contact_path(@contact.id), notice: 'Draft triage discarded.'
+    elsif params.require(:save_for_later) == 'true'
+      save_for_later(@contact.id, contact_params, contact_needs_params)
+      redirect_to new_contact_path, notice: 'Triage temporarely saved.'
```
Comment (jamiebuckley) [inline app/controllers/triage_controller.rb:25]: Small typo: "temporarily"

File: app/views/shared/_side-nav.html.erb [hunk 2]
```diff
@@ -47,6 +47,13 @@
           <% end %>
         </li>
       <% end %>
+
+      <!-- Draft Triage in progress -->
+      <% if session[:triage] %>
+      <li class="main-menu__item">
+        <%= link_to "Draft Triage (1)", contact_triage_path(session[:triage][:contact_id]), class: "main-menu__link main-menu__draft" %>
```
Comment (jamiebuckley) [inline app/views/shared/_side-nav.html.erb:54]: What does the (1) mean here?  It's currently possible to add multiple triages for different people

File: features/create_note.feature [hunk 3]
```diff
@@ -1,3 +1,4 @@
+@javascript
```
Comment (jamiebuckley) [inline features/create_note.feature:1]: Why did this one need to run with a js driver?

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -10,7 +10,13 @@
       <header class="panel content-header">
         <%= yield :header %>
         <nav class="content-header__actions-right">
-          <%= link_to "Add Enquiry", new_contact_path, {:remote => true, 'data-toggle' => "modal", 'data-target' => '#modal-window', id: "btnAddEnquiry", class: "button button--dark" } %>
+          <% if request.path_info.include?('/triage') %>
+            <!-- Ask the user to confirm you want to exit triage -->
+            <%= link_to "Add Enquiry", new_contact_path, {:remote => true, 'data-toggle' => "modal", 'data-target' => '#modal-window', id: "btnAddEnquiry", class: "button button--dark" } %>
+          <% else %>
+            <!-- Go straight to Add enquiry page -->
+            <%= link_to "Add Enquiry", new_contact_path, class: "button button--dark", id:"btnAddEnquiry" %>
```
Comment (jamiebuckley) [inline app/views/layouts/application.html.erb:18]: This should check if you have contact create permissions now

Comment (rik86future) [reply app/views/layouts/application.html.erb:18]: ok good point

File: app/views/shared/_side-nav.html.erb (hunk 2 above)
Comment (rik86future) [reply app/views/shared/_side-nav.html.erb]: yes, maybe I should remove it

File: features/create_note.feature (hunk 3 above)
Comment (rik86future) [reply features/create_note.feature:1]: I am not sure, if I run it with BROWSER=headless_chrome it all works fine for me without tagging. I think there is a lot of extra js in the applicaiton.html.erb now. I think we should just switch to headless_chrome and remove all @js tags

Comment (rik86future) [reply features/create_note.feature:1]: it says it does not find the needs table...

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (rik86future) [reply app/views/layouts/application.html.erb:19]: done

File: app/views/shared/_side-nav.html.erb (hunk 2 above)
Comment (rik86future) [reply app/views/shared/_side-nav.html.erb]: done

File: app/controllers/triage_controller.rb (hunk 1 above)
Comment (rik86future) [reply app/controllers/triage_controller.rb]: thanks, done

File: features/create_note.feature (hunk 3 above)
Comment (jamiebuckley) [reply features/create_note.feature:1]: Ah, I think it's because for the notes, it runs through the triage form

Comment (jamiebuckley) [reply features/create_note.feature:1]: This passes for me if you add:
```
page.fill_in 'contact_needs_needs_list_0_start_on', with: '01/01/2020' if need == 'Phone triage'
```
to need_steps.rb "I add needs {string}"

Comment (jamiebuckley) [reply features/create_note.feature:1]: Without javascript there isn't a default date option set

Comment (jamiebuckley) [reply features/create_note.feature:1]: The rack tests run about 5-10x quicker than headless_chrome, so there's an advantage in avoiding that.
Given that we are using this more as an e2e test suite than an acceptance test suite, there's an advantage in that imo.

File: app/views/contacts/_form_fields.html.erb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 <fieldset class="field-section field-section--two-cols">
   <div class="field">
     <%= form.label :first_name, class: "field__label" %>
-    <%= form.text_field :first_name %>
+    <%= form.text_field :first_name, id: "first-name" %>
```
Comment (jamiebuckley) [inline app/views/contacts/_form_fields.html.erb:6]: This is what's breaking the cucumber tests atm btw, if it saves you time

Comment (jamiebuckley) [reply app/views/contacts/_form_fields.html.erb:6]: person_steps is looking for 'contact_first_name'

**Comment by jamiebuckley** [comment]
I'll approve and leave it up to you whether to remove the @js flag with the suggestion above - it's a minor point anyway.  Cucumber tests are all green when the 'contact_first_name' => 'first-name' ids are fixed.

File: app/views/contacts/_form_fields.html.erb (hunk 5 above)
Comment (rik86future) [reply app/views/contacts/_form_fields.html.erb:6]: ah dear

**Comment by rik86future** [comment]
> I'll approve and leave it up to you whether to remove the @js flag with the suggestion above - it's a minor point anyway. Cucumber tests are all green when the 'contact_first_name' => 'first-name' ids are fixed.

Thanks for this, I have made a test try using headless_chrome, it takes more or less the same time, slightly less, but I will revert anyway

**Comment by rik86future** [comment]
[12. Quick capture referral](https://trello.com/c/TGmpxL1m/359-12-quick-capture-referral)

**Pull request merged**